2/DH_multiprecision.cpp

```diff
@@ -0,0 +1,110 @@
+#include <iostream>
+#include <random>
+#include <boost/multiprecision/cpp_int.hpp>
+#include <boost/multiprecision/miller_rabin.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <boost/random.hpp>
+
+
+using namespace boost::multiprecision;
+using namespace boost::random;
+using std::cin;
+using std::cout;
+using std::endl;
+using std::pair;
+typedef cpp_int data;
+#define ITER 25
```

Comment:
```suggestion
const ITER = 25;
```
Лучше использовать такой синтаксис вместо `define`.